tests/views/test_collection_exercise.py

```diff
@@ -2075,16 +2080,379 @@ def test_survey_edit_permission(self, mock_request):
+        mock_request.get(url_ces_by_survey, json=self.collection_exercises)
+        mock_request.get(url_ce_by_id, json=collection_exercise_details["collection_exercise"])
+        mock_request.get(url_get_collection_exercise_events, json=self.collection_exercise_events)
+        mock_request.get(
+            f"{url_get_collection_instrument}?{ci_search_string}", json=self.collection_instruments, complete_qs=True
+        )
+        mock_request.get(
+            f"{url_get_collection_instrument}?{ci_type_search_string_eq}", json=self.eq_ci_selectors, complete_qs=True
+        )
+        mock_request.get(url_link_sample, json=[sample_summary_id])
+        mock_request.get(url_get_sample_summary, json=self.sample_summary)
+        mock_request.get(url_get_classifier_type_selectors, json=classifier_type_selectors)
+        mock_request.get(url_get_classifier_type, json=classifier_types)
+
+        response = self.client.get(f"/surveys/{short_name}/{period}/view-sample-ci")
+
+        self.assertEqual(200, response.status_code)
+        self.assertIn("Sample loaded".encode(), response.data)
+        self.assertIn("Replace sample file".encode(), response.data)
+        self.assertIn("SEFT collection instruments".encode(), response.data)
+        self.assertIn("0".encode(), response.data)
+        self.assertIn("Upload SEFT files".encode(), response.data)
+        self.assertIn("Done".encode(), response.data)
+
+    @requests_mock.mock()
+    def test_seft_loaded_sample_view_sample_ci_page_no_survey_permission(self, mock_request):
+        mock_request.get(url_get_survey_by_short_name, json=self.survey)
+        mock_request.get(url_ces_by_survey, json=self.collection_exercises)
+        mock_request.get(url_ce_by_id, json=collection_exercise_details["collection_exercise"])
+        mock_request.get(url_get_collection_exercise_events, json=self.collection_exercise_events)
+        mock_request.get(
+            f"{url_get_collection_instrument}?{ci_search_string}", json=self.collection_instruments, complete_qs=True
+        )
+        mock_request.get(
+            f"{url_get_collection_instrument}?{ci_type_search_string_eq}", json=self.eq_ci_selectors, complete_qs=True
+        )
+        mock_request.get(url_link_sample, json=[sample_summary_id])
+        mock_request.get(url_get_sample_summary, json=self.sample_summary)
+        mock_request.get(url_get_classifier_type_selectors, json=classifier_type_selectors)
+        mock_request.get(url_get_classifier_type, json=classifier_types)
+
+        response = self.client.get(f"/surveys/{short_name}/{period}/view-sample-ci")
+
+        self.assertEqual(200, response.status_code)
+        self.assertIn("Sample loaded".encode(), response.data)
+        self.assertNotIn("Replace sample file".encode(), response.data)
+        self.assertIn("SEFT collection instruments".encode(), response.data)
+        self.assertIn("0".encode(), response.data)
```

Comment:
:see_no_evil: I don't think we can assert against a single number with nothing else. This is done in a number of places

Reply:
Removed the single number assertions that I added, didn't do it for older tests as there might be some actual reason for them